integration-pod-api-client/src/test/java/org/symphonyoss/integration/pod/api/client/IntegrationAuthApiClientTest.java

```diff
@@ -77,62 +88,113 @@ public void testGetUserAuthDataNullUrl() throws RemoteApiException {
 
     UserAuthorizationData data = new UserAuthorizationData();
-    doReturn(data).when(httpClient).doGet(path, headerParams, queryParams, UserAuthorizationData.class);
+    apiClient.saveUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, data);
+  }
 
-    assertEquals(data, apiClient.getUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, MOCK_USER_ID, MOCK_URL));
+  @Test(expected = ForbiddenAuthException.class)
+  public void testSaveUserAuthDataForbidden() throws RemoteApiException {
+    RemoteApiException apiException = new RemoteApiException(403, "forbidden");
+    mockRemoteException(TestTypeEnum.SAVE, apiException);
+
+    UserAuthorizationData data = new UserAuthorizationData();
+    apiClient.saveUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, data);
   }
 
-  private void mockRemoteException(RemoteApiException apiException) throws RemoteApiException {
-    String path = "/v1/configuration" + MOCK_INTEGRATION_ID + "/auth/user";
+  @Test
+  public void testSaveUserAuthData() throws RemoteApiException {
+    UserAuthorizationData data = new UserAuthorizationData();
+    apiClient.saveUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, data);
+  }
 
-    Map<String, String> headerParams = new HashMap<>();
-    headerParams.put(SESSION_TOKEN_HEADER_PARAM, MOCK_SESSION);
+  @Test(expected = UnauthorizedUserException.class)
+  public void testSearchUserAuthDataUnauthorized() throws RemoteApiException {
+    RemoteApiException apiException = new RemoteApiException(401, "unauthorized");
+    mockRemoteException(TestTypeEnum.SEARCH, apiException);
+
+    apiClient.searchUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, null);
+  }
 
-    Map<String, String> queryParams = new HashMap<>();
-    queryParams.put(USER_ID, String.valueOf(MOCK_USER_ID));
-    queryParams.put(URL, MOCK_URL);
+  @Test(expected = ForbiddenAuthException.class)
+  public void testSearchUserAuthDataForbidden() throws RemoteApiException {
+    RemoteApiException apiException = new RemoteApiException(403, "forbidden");
+    mockRemoteException(TestTypeEnum.SEARCH, apiException);
+
+    apiClient.searchUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, null);
+  }
+
+  @Test
+  public void testSearchUserAuthData() throws RemoteApiException {
+    Map<String, String> filter = new HashMap<>();
+    UserAuthorizationDataList expected = new UserAuthorizationDataList();
+    expected.add(new UserAuthorizationData());
+    doReturn(expected).when(httpClient).doGet(
+        anyString(), anyMap(), anyMap(), eq(UserAuthorizationDataList.class));
+
+    List result = apiClient.searchUserAuthData(MOCK_SESSION, MOCK_INTEGRATION_ID, filter);
+    assertEquals(expected, result);
+  }
 
-    doThrow(apiException).when(httpClient).doGet(path, headerParams, queryParams, UserAuthorizationData.class);
+  private void mockRemoteException(TestTypeEnum testType, RemoteApiException apiException)
+      throws RemoteApiException {
+
+    if (testType == TestTypeEnum.FIND) {
```

Comment:
It looks like this method is doing 3 different things. Split this code into 3 methods: mockRemoteExceptionToFind, mockRemoteExceptionToSearch, and mockRemoteExceptionToSearch.

If you can use real values rather than any() it would be better.